addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -23,6 +24,12 @@ class ChannelMember(models.Model):
     message_unread_counter = fields.Integer('Unread Messages Counter', compute='_compute_message_unread', compute_sudo=True)
     fold_state = fields.Selection([('open', 'Open'), ('folded', 'Folded'), ('closed', 'Closed')], string='Conversation Fold State', default='open')
     is_minimized = fields.Boolean("Conversation is minimized")
+    custom_notifications = fields.Selection(
+        [("mentions", "Mentions Only"), ("no_notif", "Nothing")],
+        "Customized Notifications",
+        help="All Messages if not specified",
+    )
```

Comment:
review note: renamed the field and changed its behavior so that the default is empty
This will avoid polluting the database for the basing case, and prevents from having to create an upgrade script to set value to "all" for everyone